Copy results files fail

Occasionally, the runfarm instance will fail to copy files back to the manager. Manually re-running the commands works. See [here](https://groups.google.com/forum/#!topic/firesim/GI29vu-93es) for the initial report.

Here is a trace provided by hzhuo2@wisc.edu:
2019-08-01 21:23:10,886 [instance_log] [INFO ]  [192.168.1.43] Slot 0 completed! copying results.
2019-08-01 21:23:10,886 [flush       ] [DEBUG]  [localhost] local: mkdir -p /home/centos/firesim/deploy/results-workload/2019-08-01--21-04-56-cus-fed29//cus-fed29-test/
2019-08-01 21:23:10,893 [copy_back_jo] [DEBUG]  [localhost] 
2019-08-01 21:23:10,893 [copy_back_jo] [DEBUG]  [localhost] 
2019-08-01 21:23:10,894 [flush       ] [DEBUG]  [192.168.1.43] run: sudo mkdir -p /home/centos/sim_slot_0/mountpoint
2019-08-01 21:23:11,063 [flush       ] [DEBUG]  [192.168.1.43] run: sudo mount /home/centos/sim_slot_0/cus-fed29-test.img /home/centos/sim_slot_0/mountpoint
2019-08-01 21:23:11,600 [flush       ] [DEBUG]  [192.168.1.43] run: sudo chmod -Rf 777 /home/centos/sim_slot_0/mountpoint
2019-08-01 21:23:13,551 [flush       ] [DEBUG]  Fatal error: run() received nonzero return code 1 while executing!
2019-08-01 21:23:13,551 [flush       ] [DEBUG]  Requested: sudo chmod -Rf 777 /home/centos/sim_slot_0/mountpoint
2019-08-01 21:23:13,551 [flush       ] [DEBUG]  Executed: /bin/bash -l -c "sudo chmod -Rf 777 /home/centos/sim_slot_0/mountpoint"
2019-08-01 21:23:13,551 [flush       ] [DEBUG]  Aborting.
2019-08-01 21:23:13,573 [<module>    ] [ERROR]  Fatal error.

--- a/deploy/runtools/firesim_topology_elements.py
+++ b/deploy/runtools/firesim_topology_elements.py
@@ -241,7 +241,7 @@ class FireSimServerNode(FireSimNode):
         runcommand = self.server_hardware_config.get_boot_simulation_command(
             slotno, all_macs, all_rootfses, all_linklatencies, all_maxbws,
             self.server_profile_interval, all_bootbins, self.trace_enable,
-            self.trace_select, self.trace_start, self.trace_end, 
+            self.trace_select, self.trace_start, self.trace_end,
             self.autocounter_readrate, all_shmemportnames)
 
         run(runcommand)
@@ -279,7 +279,8 @@ class FireSimServerNode(FireSimNode):
                     rfsname = """/home/centos/sim_slot_{}/{}""".format(simserverindex, rfsname)
 
                 run("""sudo mount {blockfile} {mntpt}""".format(blockfile=rfsname, mntpt=mountpoint))
-                run("""sudo chmod -Rf 777 {}""".format(mountpoint))
+                run("""sudo chattr -i {}/etc/sysconfig/nfs""".format(mountpoint))
+                run("""sudo chown -R centos {}""".format(mountpoint))
 
             ## copy back files from inside the rootfs
             with warn_only(), StreamLogger('stdout'), StreamLogger('stderr'):

--- a/deploy/runtools/firesim_topology_elements.py
+++ b/deploy/runtools/firesim_topology_elements.py
@@ -279,7 +279,7 @@ class FireSimServerNode(FireSimNode):
                     rfsname = """/home/centos/sim_slot_{}/{}""".format(simserverindex, rfsname)
 
                 run("""sudo mount {blockfile} {mntpt}""".format(blockfile=rfsname, mntpt=mountpoint))
-                run("""sudo chattr -i {}/etc/sysconfig/nfs""".format(mountpoint))
+                run("""sudo chattr -R -i {}""".format(mountpoint))
                 run("""sudo chown -R centos {}""".format(mountpoint))
 
             ## copy back files from inside the rootfs

